TERMS.md

```diff
@@ -0,0 +1,121 @@
+## Service
+	- (General) A service is a software functionality or set of functionalities that can be activated by a client. A key characteristic of a service in support of SOA is that it has low coupling to other services and strives to provide a useful function also if used independently from other services.
+	- (VSC) A VSC **Service** contains one or more **Interface**s, which are defined or referenced within the service definition.  The service definition may within its namespace also define **Type**s, constants and other definitions required to make use of the service.
+	- A Service name is not in itself a Namespace, but its contents must be defined within *at least one* **Namespace**.  The Namespace in question is referenced or defined from within the Service definition.
+	- A Service must have its fundamental definition in **one** single .yaml formatted file, although this file might use references to other parts.  In other words, all Interfaces that make up a service must be mentioned within this file*.
+	- A file can contain a maximum of one Service.
```

Comment:
This is no longer needed as restriction if we have an explicit service keyword, like in https://github.com/miketsukerman/vsc-tools/blob/use-dataclasses-for-ast/tests/input.yaml, right? Then you could theoretically have multiple services defined in one file, or?

Reply:
Yes it's definitely possible - we could remove this restriction if that is desired.  I deliberately put some things like that in so that we would spark a discussion. 

But whatever we choose it has consequences.  If File is equivalent to Service, then we don't need to have a service: keyword at the top of every YAML, and so the depth/indentation of the tree is less, and the file becomes is easier to write.  

(The opposite for any feature we choose to add).

Either way, we could use the word 'Service' in our internal representation, as a concept.